crates/dbsp/src/time/product.rs

```diff
@@ -79,6 +79,8 @@ where
     TOuter: DBTimestamp,
     TInner: DBTimestamp,
 {
+    type Nested = Product<Self, u32>;
```

Comment:
We use `()` for the outermost timestamp to save memory and use tricks to make sure that this is ok.  This may have to change when we introduce persistence, at which point we'll need to allocate enough bits to the timestamp so that it doesn't overflow.  For all other timestamps overflow can only happen when the number of nested iterations exceeds 2^31 or 2^32, which is unlikely.

Reply:
Because it's unlikely we should use checked add for these.